File: User Services/client/src/main/java/com/worksphere/dashboardapi/entity/Complaint.java
package com.worksphere.dashboardapi.entity;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "complaints")
public class Complaint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "complaint_id")
    private int complaintId;

    // who raised the complaint (client)
    @Column(name = "from_user_id", nullable = false)
    private int fromUserId;

    // against whom (freelancer)
    @Column(name = "to_user_id", nullable = false)
    private int toUserId;

    // for which service
    @Column(name = "service_id", nullable = false)
    private int serviceId;

    @Column(name = "description")
    private String description;

    @Column(name = "create_date")
    private LocalDate createDate;

    @Column(name = "status")
    private String status;

    public int getComplaintId() {
        return complaintId;
    }

    public void setComplaintId(int complaintId) {
        this.complaintId = complaintId;
    }

    public int getFromUserId() {
        return fromUserId;
    }

    public void setFromUserId(int fromUserId) {
        this.fromUserId = fromUserId;
    }

    public int getToUserId() {
        return toUserId;
    }

    public void setToUserId(int toUserId) {
        this.toUserId = toUserId;
    }

    public int getServiceId() {
        return serviceId;
    }

    public void setServiceId(int serviceId) {
        this.serviceId = serviceId;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDate getCreateDate() {
        return createDate;
    }

    public void setCreateDate(LocalDate createDate) {
        this.createDate = createDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: RestAPIs/Admin_Services/Models/Complaint.cs
using System;

namespace Admin_Services.Models
{
    public partial class Complaint
    {
        public int ComplaintId { get; set; }

        // ================= FOREIGN KEYS =================
        public int FromUserId { get; set; }   // Client
        public int ToUserId { get; set; }     // Freelancer
        public int ServiceId { get; set; }

        // ================= DATA =================
        public string? Description { get; set; }
        public DateOnly? CreateDate { get; set; }
        public string Status { get; set; } = string.Empty;

        // ================= NAVIGATION =================
        public User FromUser { get; set; } = null!;
        public User ToUser { get; set; } = null!;
        public Service Service { get; set; } = null!;
    }
}
